logstreams/src/main/java/io/zeebe/logstreams/state/StateController.java

```diff
@@ -192,56 +265,98 @@ public void put(long key, BufferWriter valueWriter) {
 
     try {
       db.put(dbLongBuffer.byteArray(), bytes);
-    } catch (RocksDBException e) {
+    } catch (final RocksDBException e) {
       LangUtil.rethrowUnchecked(e);
     }
   }
 
-  public void put(byte[] key, byte[] valueBuffer) {
+  public void put(final byte[] key, final byte[] valueBuffer) {
     try {
       db.put(key, valueBuffer);
-    } catch (RocksDBException e) {
+    } catch (final RocksDBException e) {
       LangUtil.rethrowUnchecked(e);
     }
   }
 
-  public void delete(long key) {
+  public void delete(final long key) {
     setKey(key);
     try {
       db.delete(dbLongBuffer.byteArray());
-    } catch (RocksDBException e) {
+    } catch (final RocksDBException e) {
       LangUtil.rethrowUnchecked(e);
     }
   }
 
-  public void delete(byte[] key) {
+  public void delete(final byte[] key) {
     try {
       db.delete(key);
-    } catch (RocksDBException e) {
+    } catch (final RocksDBException e) {
       LangUtil.rethrowUnchecked(e);
     }
   }
 
+  public int get(
```

Comment:
In both `get` method, we could check the bloom filter beforehand. If a key does not exist 100%, then we can know in advance and avoid an extra IO call. This is an optimization though, so I'm OK leaving it out for now.

Reply:
Is this really an performance improvement ? Because in the cases where the key exist we have every time to do two calls right?

Reply:
Checking the bloom filter is in memory though, so it's pretty fast, and it avoids doing an IO call in the case where the key isn't there. In some cases it can even fetch the value from the cache so you avoid IO calls as well when it's there, but it returns a StringBuilder which was kind of annoying to extract values from.

I would say an additional quick check in mem to avoid a potential IO call is a better trade off than always doing the IO call, but since it's optimization I'm fine not doing it yet.

Reply:
Can you create a new issue for this?